### ISSUE TITLE
[Mat Table] Voiceover is not announcing mat-table column count correctly in forward navigation.

### ISSUE THREAD

**Issue by bq007** [angular/components#22413]
#### Description

When forward navigating to Mat Table, the VoiceOver screen speaker announces table columns as 0. 
![Screen Shot for mat table bug](https://user-images.githubusercontent.com/67595364/113763890-39216100-96e8-11eb-96cc-91a9788843f0.png)

#### Reproduction
Use Angular Material demos for mat table to reproduce the issue: 
https://material.angular.io/components/table/examples#table-dynamic-columns

Steps to reproduce:
1. Enable VoiceOver screen speaker. 
2. Open [Angular Material demos for mat table](https://material.angular.io/components/table/examples#table-dynamic-columns)
3. Forward navigate to the the main table of "Table dynamically changing the columns displayed" demo by Caps lock+Arrow keys.
4. Listen to VoiceOver screen speaker announcing the table columns count.  
 
 
#### Expected Behavior

The VoiceOver screen speaker should announce the table columns count correctly.

#### Actual Behavior

The VoiceOver screen speaker should announce the table columns count as 0 --"Table 0 column 11 rows".
Even manually adding ["aria-colcount"](https://www.digitala11y.com/aria-colcountproperties/) doesn't fix the issue. 

#### Environment

- AngularJS: 10.0.0
- AngularJS Material: 10.0.0
- OS: macOS Big Sur Version 11.2.1
- Browsers: Chrome Version 89.0.4389.90 (Official Build) (x86_64)

**Comment by zarend** [comment]
Looks like this already got fixed.
<img width="1036" alt="Screen Shot 2022-04-11 at 1 33 08 PM" src="https://user-images.githubusercontent.com/7720245/162827370-291b0d64-a09a-4442-851d-9a49b809824c.png">

I tested with
 - Chrome Version 100.0.4896.75 (Official Build) (x86_64)
 - MacOS 12.3 (21E230)
